tests/test_Histogram.py

```diff
@@ -0,0 +1,133 @@
+
+def test_histogram_creation_with_tuple():
+    # Test histogram creation with a tuple
+    hist = Histogram((0, 10, 10))
+    assert hist.number_of_bins_ == 10
+    assert np.allclose(hist.bin_edges_, np.linspace(0, 10, num=11))
+
+def test_histogram_creation_with_list():
+    # Test histogram creation with a list
+    hist = Histogram([0, 2, 4, 6, 8, 10])
+    assert hist.number_of_bins_ == 5
+    assert np.allclose(hist.bin_edges_, np.array([0, 2, 4, 6, 8, 10]))
+
+def test_histogram_creation_with_invalid_input():
+    # Test histogram creation with invalid input
+    with pytest.raises(TypeError):
+        # Passing incorrect input type
+        hist = Histogram("invalid_input")
+
+    with pytest.raises(ValueError):
+        # Passing tuple with invalid values
+        hist = Histogram((10, 0, 10))  # hist_min > hist_max
+    
+    with pytest.raises(ValueError):
+        # Passing tuple with non-integer number of bins
+        hist = Histogram((0, 10, 1.4))
+    
+    with pytest.raises(ValueError):
+        hist = Histogram((0, 10, -1)) # num_bins <= 0
+
+def test_set_error_with_invalid_input():
+    # Test setting error with invalid input
+    hist = Histogram((0, 10, 10))
+    with pytest.raises(ValueError):
+        # Passing incorrect length of error list
+        hist.set_error([1, 2, 3])
+
+def test_set_systematic_error_with_invalid_input():
+    # Test setting systematic error with invalid input
+    hist = Histogram((0, 10, 10))
+    with pytest.raises(ValueError):
+        # Passing incorrect length of systematic error list
+        hist.set_systematic_error([1, 2, 3])
+
+def test_add_value_single_number():
+    # Test adding a single number to the histogram
+    hist = Histogram((0, 10, 10))
+    hist.add_value(4.5)
+    assert np.allclose(hist.histogram(), np.array([0, 0, 0, 0, 1, 0, 0, 0, 0, 0]))
+
+def test_add_value_list():
+    # Test adding a list of numbers to the histogram
+    hist = Histogram((0, 10, 10))
+    hist.add_value([1, 3, 5, 7, 9])
+    assert np.allclose(hist.histogram(), np.array([0, 1, 0, 1, 0, 1, 0, 1, 0, 1]))
+
+def test_average():
+    # Test averaging histograms
+    hist = Histogram((0, 10, 10))
+    hist.add_value([1, 3, 5, 7, 9])
+    hist.add_histogram()
+    hist.add_value([2, 4, 6, 8, 9])
+    hist.average()
+    assert np.allclose(hist.histogram(), np.array([0, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 1.0]))
+
+def test_write_to_file(tmp_path):
+    # Test writing histograms to a file
+    hist = Histogram((0, 4, 4))
+    hist.add_value([1, 2, 3])
+    hist.statistical_error()
+    hist_labels = [{'bin_center': 'Bin Center', 'bin_low': 'Bin Low', 'bin_high': 'Bin High',
+                    'distribution': 'Distribution', 'stat_err+': 'Stat Error+', 'stat_err-': 'Stat Error-',
+                    'sys_err+': 'Sys Error+', 'sys_err-': 'Sys Error-'}]
+    filename = tmp_path / "test_histograms.csv"
+    hist.write_to_file(filename, hist_labels)
+
+    print(hist.histogram())
```

Comment:
Should tests really print out every time they run?

Reply:
You are completely right. This was a test of the test leftover.